app/main/views/signup.py

```diff
@@ -210,3 +215,29 @@ def application_update(id, step=None):
         return jsonify(result)
     else:
         return redirect(url_for('.render_application', id=id, step=form_data['next_step_slug']))
+
+
+@main.route('/application/<int:id>/documents/<slug>', methods=['GET'])
+@applicant_login_required
+def download_single_file(id, slug):
+    application = data_api_client.get_application(id)
+    if not can_user_view_application(application):
+        abort(403, 'Not authorised to access application')
+    if is_application_submitted(application):
+        abort(400, 'Application already submitted')
+
+    file = s3_download_file(slug, os.path.join(S3_PATH, str(id)))
+
+    return Response(file, mimetype='binary/octet-stream')
```

Comment:
Would be nice to set this to the corresponding mimetype of the file extension, which you can get with: 

import mimetypes
mimetype = mimetypes.guess_type(filename)[0] or 'binary/octet-stream'